src/sentry/ownership/grammar.py

```diff
@@ -35,10 +35,13 @@
 
 ownership = line*
 
-line = _ (comment / rule / empty) newline?
+line = _ (comment / rule / rule_comment / empty) newline?
 
 rule = _ matcher owners
 
+# A partial rule is a rule without owners
+rule_comment = _ matcher comment?
```

Comment:
One more edge case:

> Escaping a pattern starting with # using \ so it is treated as a pattern and not a comment

`/apps/\#1 @user` should get parsed as `/apps/#1 @user` and not as though it has a trailing comment